vm.c

```diff
@@ -2544,7 +2551,9 @@ rb_execution_context_mark(const rb_execution_context_t *ec)
 
     RUBY_MARK_UNLESS_NULL(ec->errinfo);
     RUBY_MARK_UNLESS_NULL(ec->root_svar);
-    rb_mark_tbl(ec->local_storage);
+    if (ec->local_storage) {
+        rb_id_table_foreach_values(ec->local_storage, mark_local_storage_i, NULL);
```

Comment:
Would it make more sense to add `rb_id_table_mark`?

Reply:
I think not as ID tables rarely have VALUE types as values and tend to hold more complex structs (as opposed to `st_table` which generally tends to map to VALUEs more often).

See for example:

* https://github.com/ruby/ruby/blob/0c2d731ef210c9121e2a97cc5b0d7594a80389f3/variable.c#L428-L449
* https://github.com/ruby/ruby/blob/e082f41611755b0fde967fccf3174c90ecb8469e/gc.c#L2515-L2528
* https://github.com/ruby/ruby/blob/e082f41611755b0fde967fccf3174c90ecb8469e/gc.c#L4955-L4971

Reply:
Okay, fair enough. If you don't think we are repeating the same code over and over again, I'm okay with this approach.